go/client/json_api_common.go

```diff
@@ -197,6 +197,11 @@ func (c *cmdAPI) decode(ctx context.Context, r io.Reader, w io.Writer, h handler
 			return err
 		}
 
+		defer func() {
```

Comment:
the `defer` should be defined outside of the loop, otherwise the function will be scheduled on each loop iteration